modules/lib/lib-grid/src/main/resources/lib/xp/grid.ts

```diff
@@ -125,29 +153,25 @@ export class SharedMap {
      *
      * @param {string} key the key whose associated value is to be removed
      */
-    delete(key: string): void {
+    delete(key: keyof Map): void {
         this.map.delete(key);
     }
 }
 
-interface SharedMapHandler {
-    getMap(mapId: string): ISharedMap;
-}
-
-function requireNotNull<T extends string | SharedMapModifierFn>(value: T, parameterName: string): T {
+function requireNotNull<T>(value: T, parameterName: string): T {
     if (value == null) {
         throw `Parameter "${parameterName}" is required`;
     }
     return value;
 }
 
-function convertValue(value: SharedMapValueType): SharedMapValueType {
+function convertValue<Map extends GridMap, Key extends keyof Map>(value: Map[Key] | null): Map[Key] | null {
```

Comment:
When I see the `as` keyword, I always try to figure out a way to solve a problem with out it.

In this case I think we might be fooling ourselves a little bit with the return type of `convertValue()`. In reality it can return `Record<string, unknown>` from `getMap()` or `object[]` from `getList()` too.

The job of this function is to create values that can be consumed by `JavaSharedMap`. So I think we should update the return type here, and change `JavaSharedMap` to accept the actual types returned.

```suggestion
function convertValue(value: SharedMapValueType | null): SharedMapValueType | Record<string, unknown> | object[] | null {
```

---

Not now, but in the long run having "branded types" to represent the Java-version of values might help prevent bugs. Because developers will not be able to use JavaScript-values, when Java-values are needed.